crates/dojo-bindgen/src/plugins/unity/mod.rs

```diff
@@ -1,30 +1,334 @@
+
+    fn format_model(model: &Composite) -> String {
+        let fields = model
+            .inners
+            .iter()
+            .map(|field| {
+                format!(
+                    "[ModelField(\"{}\")]\n    public {} {};",
+                    field.name,
+                    UnityPlugin::map_type(field.token.type_name().as_str()),
+                    field.name,
+                )
+            })
+            .collect::<Vec<String>>()
+            .join("\n\n    ");
+
+        format!(
+            "
+// Model definition for `{}` model
+public class {} : ModelInstance {{
+    {}
+
+    // Start is called before the first frame update
+    void Start() {{
+    }}
+
+    // Update is called once per frame
+    void Update() {{
+    }}
+}}
+        ",
+            model.type_path,
+            model.type_name(),
+            fields
+        )
+    }
+
+    fn handle_model(&self, model: &DojoModel, handled_tokens: &mut Vec<Composite>) -> String {
+        let mut out = String::new();
+        out += UnityPlugin::generated_header().as_str();
+        out += "using System;\n";
+        out += "using Dojo;\n";
+        out += "using Dojo.Starknet;\n";
+
+        let mut model_struct: Option<&Composite> = None;
+        let tokens = &model.tokens;
+        for token in &tokens.structs {
+            handled_tokens.push(token.to_composite().unwrap().to_owned());
+
+            // first index is our model struct
+            if token.type_name() == model.name {
+                model_struct = Some(token.to_composite().unwrap());
+                continue;
+            }
+
+            out += UnityPlugin::format_struct(token.to_composite().unwrap()).as_str();
+        }
+
+        for token in &tokens.enums {
+            handled_tokens.push(token.to_composite().unwrap().to_owned());
+            out += UnityPlugin::format_enum(token.to_composite().unwrap()).as_str();
+        }
+
+        out += "\n";
+
+        out += UnityPlugin::format_model(model_struct.expect("model struct not found")).as_str();
+
+        out
+    }
+
+    fn format_system(system: &Function, handled_tokens: &[Composite]) -> String {
+        let args = system
+            .inputs
+            .iter()
+            .map(|arg| format!("{} {}", UnityPlugin::map_type(&arg.1.type_name()), arg.0,))
+            .collect::<Vec<String>>()
+            .join(", ");
+
+        let calldata = system
+            .inputs
+            .iter()
+            .map(|arg| {
+                let token = arg.1.to_composite().unwrap();
+                // r#type doesnt seem to be working rn.
+                // match token.unwrap().r#type {
+                //     CompositeType::Struct => {
+                //         // doesnt support structs yet
+                //         format!("new FieldElement({})", arg.0)
+                //     }
+                //     _ => {
+                //         format!("new FieldElement({})", arg.0)
+                //     }
+                // }
```

Comment:
Can this be cleaned up if not used for now? And we keep the comment for later uses?